app/src/main/java/tttTDD/Java/Board.java

```diff
@@ -1,44 +1,90 @@
 package tttTDD.Java;
 
-import org.checkerframework.common.value.qual.IntRange;
-
-import java.lang.reflect.Array;
+import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 public class Board {
+    private final int DEFAULT_SIZE = 3;
     private final String[] board;
+    private int numberOfSpaces;
+    private int[][] winningSets = {
+            {0, 1, 2},
+            {3, 4, 5},
+            {6, 7, 8},
+            {0, 3, 6},
+            {1, 4, 7},
+            {2, 5, 8},
+            {0, 4, 8},
+            {2, 4, 6},
+    };
 
     public Board() {
-        this.board = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+        this.board = initializeBoard(DEFAULT_SIZE);
     }
 
     public Board(int boardSize) {
-        this.board = new String[boardSize * boardSize];
-        for (int i = 0; i < this.board.length; i++) {
-            this.board[i] = String.valueOf(i + 1);
-        }
+        this.board = initializeBoard(boardSize);
     }
 
-    public String[] getBoard() {
-        return this.board;
+    public String getSpace(int location) {
+        return isLocationValid(location) ? null : this.board[location];
     }
```

Comment:
Intentionally returning null values increases the risk of null pointer exceptions, because it's pretty common for consumers to forget that the method could always return (e.g.) int _or null_.

A fairly modern way to handle this is to use the [Optional single-value container](https://www.oracle.com/technical-resources/articles/java/java8-optional.html) introduced in Java 8. This would require the consumer to unwrap it to get to the value, forcing them to consider the negative case.

A simpler way would be to throw a (user-defined) exception. Consumers might ignore the exception annotation and naively use the returned value, but if so the null value will pass the exception up the stack until it halts the program, focusing attention on the buggy code, rather than creating hard to diagnose problems in methods that consume this.